pkg/registry/core/node/storage/storage_test.go

```diff
@@ -165,127 +187,49 @@ func TestShortNames(t *testing.T) {
+	type testCase struct {
 		name     string
-		node     api.Node
+		node     *api.Node
 		query    string
 		location string
 		err      error
-	}{
-		{
-			name: "proxyable hostname with default port",
-			node: api.Node{
-				ObjectMeta: metav1.ObjectMeta{Name: "node0"},
-				Status: api.NodeStatus{
-					Addresses: []api.NodeAddress{
-						{
-							Type:    api.NodeInternalIP,
-							Address: "10.0.0.1",
-						},
-					},
-				},
-			},
-			query:    "node0",
-			location: "10.0.0.1:10250",
-			err:      nil,
-		},
-		{
-			name: "proxyable hostname with kubelet port in query",
-			node: api.Node{
-				ObjectMeta: metav1.ObjectMeta{Name: "node0"},
-				Status: api.NodeStatus{
-					Addresses: []api.NodeAddress{
-						{
-							Type:    api.NodeInternalIP,
-							Address: "10.0.0.1",
-						},
-					},
-				},
-			},
-			query:    "node0:5000",
-			location: "10.0.0.1:5000",
-			err:      nil,
-		},
-		{
-			name: "proxyable hostname with kubelet port in status",
-			node: api.Node{
-				ObjectMeta: metav1.ObjectMeta{Name: "node0"},
-				Status: api.NodeStatus{
-					Addresses: []api.NodeAddress{
-						{
-							Type:    api.NodeInternalIP,
-							Address: "10.0.0.1",
-						},
-					},
-					DaemonEndpoints: api.NodeDaemonEndpoints{
-						KubeletEndpoint: api.DaemonEndpoint{
-							Port: 5000,
-						},
-					},
-				},
-			},
-			query:    "node0",
-			location: "10.0.0.1:5000",
-			err:      nil,
-		},
-		{
-			name: "non-proxyable hostname with default port",
-			node: api.Node{
-				ObjectMeta: metav1.ObjectMeta{Name: "node0"},
-				Status: api.NodeStatus{
-					Addresses: []api.NodeAddress{
-						{
-							Type:    api.NodeInternalIP,
-							Address: "127.0.0.1",
-						},
-					},
-				},
-			},
-			query:    "node0",
-			location: "",
-			err:      proxyutil.ErrAddressNotAllowed,
-		},
-		{
-			name: "non-proxyable hostname with kubelet port in query",
-			node: api.Node{
-				ObjectMeta: metav1.ObjectMeta{Name: "node0"},
-				Status: api.NodeStatus{
-					Addresses: []api.NodeAddress{
-						{
-							Type:    api.NodeInternalIP,
-							Address: "127.0.0.1",
-						},
-					},
-				},
-			},
-			query:    "node0:5000",
-			location: "",
-			err:      proxyutil.ErrAddressNotAllowed,
-		},
-		{
-			name: "non-proxyable hostname with kubelet port in status",
-			node: api.Node{
-				ObjectMeta: metav1.ObjectMeta{Name: "node0"},
-				Status: api.NodeStatus{
-					Addresses: []api.NodeAddress{
-						{
-							Type:    api.NodeInternalIP,
-							Address: "127.0.0.1",
-						},
-					},
-					DaemonEndpoints: api.NodeDaemonEndpoints{
-						KubeletEndpoint: api.DaemonEndpoint{
-							Port: 443,
-						},
-					},
-				},
-			},
-			query:    "node0",
-			location: "",
-			err:      proxyutil.ErrAddressNotAllowed,
-		},
 	}
 
+	testCases := []testCase{{
+		name:     "proxyable hostname with default port",
+		node:     newNode("node0", setNodeIPAddress("10.0.0.1")),
+		query:    "node0",
+		location: "10.0.0.1:10250",
```

Comment:
nit: naming is a bit confusing to me as below code compares this value with location.Host. I'd suggest to call it `host` to avoid confusion with location object returned by redirector.ResourceLocation.
Otherwise lgtm.

Reply:
This is a good comment.  I like it :)

Reply:
Done.